### DISCUSSION TITLE
use app router without i18n routing get an error when build

### DISCUSSION THREAD

**Discussion by aurora-lch** [amannn/next-intl#1325] [Q&A]
### Description

use this example
https://github.com/amannn/next-intl/blob/main/examples/example-app-router-without-i18n-routing/src/i18n/request.ts

Error: `headers` was called outside a request scope. Read more: https://nextjs.org/docs/messages/next-dynamic-api-wrong-context

### Verifications

- [x] I've verified that the problem I'm experiencing isn't covered in [the docs](https://next-intl-docs.vercel.app/).
- [X] I've searched for similar, existing issues on GitHub and [Stack Overflow](https://stackoverflow.com/search?q=next-intl).
- [X] I've compared my app to [a working example](https://next-intl-docs.vercel.app/examples) to look for differences.

### Reproduction description

Steps to reproduce:
1. Open reproduction
2. Click on …
3. See error: … 

### Expected behaviour

not error

**Comment by amannn** [comment]
Moving this to a discussion since this seems to be a usage question regarding Next.js itself.

**Reply by alexbuznik-stark** [reply]
This seems like a basic case, I'm also having this with `requestLocale`, see [here](https://github.com/alexbuznik-stark/next-intl-demo-3/blob/9542973a332152fd57c96625c3c0f22ee585b5f9/src/i18n/request.ts#L7)

**Comment by NicholasG04** [comment]
In case it helps anyone else, I spent a while today tracking down this error - not from the example, but my own code.

I was using
```ts
const t = await getTranslations('whatever');
```

in the outer scope of my layout.tsx for metadata. You can't do this, because any call to getTranslations relies on the request scope being there for the call to `headers()` to work, and thus `locale` to be passed in tern into `getRequestConfig`.

Instead, you must use a `generateMetadata` function that runs on every request.

**Comment by alexbuznik-stark** [comment]
I'm getting this error, trying to reproduce the issue with missing RSC payload (https://github.com/amannn/next-intl/discussions/1246)

https://stackblitz.com/~/github.com/alexbuznik-stark/next-intl-demo-3
